sdk/ml/azure-ai-ml/azure/ai/ml/operations/_batch_deployment_operations.py

```diff
@@ -237,3 +252,46 @@ def _get_workspace_location(self) -> str:
         """Get the workspace location TODO[TASK 1260265]: can we cache this
         information and only refresh when the operation_scope is changed?"""
         return self._all_operations.all_operations[AzureMLResourceType.WORKSPACE].get(self._workspace_name).location
+
+    def _validate_component(self, deployment: Deployment) -> Component:
+        """Validates that the value provided is associated to an existing component
+        or otherwise we will try to create an anonymous component that will be use
+        for batch deployment
+        :param deployment: Batch deployment
+        :type deployment: ~azure.ai.ml.entities._deployment.deployment.Deployment
+        :raises ~azure.ai.ml.exceptions.ValidationException: Raised if BatchDeployment cannot be
+            successfully validated. Details will be provided in the error message.
+        :return: Component entity associated to the values provided in yaml file
+        :rtype: ~azure.ai.ml.entities.Component
+        """
+        component = None
+        if isinstance(deployment.job_definition.component, PipelineComponent):
```

Comment:
If object is component -> make a POST/PUT api call and create
else if str -> only fetch or throw

I'm not sure how the rest of our SDK behaves but it's non-obvious to me that the type of the object I'm passing will dictate what's going to happen. I would have thought that regardless, by the end of this function call my expectations are same. That this thing is guaranteed to exist (assuming I passed valid data). 
If we want this behavior split, should we have a "create_if_not_exist" flag?

Reply:
This behavior is the one specified in the spec. 
Because this is coming from a CLI request we will load the properties from the YAML file provided.
So, depending on those values we will create a "PipelineComponent" entity, or we will keep the `str` if it starts with `azureml:` prefix. 

I would suggest not adding a flag because we do have similar behavior for other scenarios using:
`upload_dependencies` function

https://github.com/Azure/azure-sdk-for-python/blob/97e1e7b0f5e52af09ea83721aecfe98816e9f4b0/sdk/ml/azure-ai-ml/azure/ai/ml/_utils/_endpoint_utils.py#L130

Reply:
Mmm, I still think that that is a bit non-obvious, but if there are other instances of this pattern then at least we are consistent. And this is a `_` prefix function so at least it's not customer facing.